pkg/cli/whoami/whoami.go

```diff
@@ -68,7 +71,7 @@ func NewCmdWhoAmI(f kcmdutil.Factory, streams genericiooptions.IOStreams) *cobra
 		Run: func(cmd *cobra.Command, args []string) {
 			kcmdutil.CheckErr(o.Complete(f))
 			kcmdutil.CheckErr(o.Validate())
-			kcmdutil.CheckErr(o.Run())
+			ocmdhelpers.CheckOAuthDisabledErr(o.Run(), o.DiscoveryClient)
```

Comment:
no need to add this check in here because we will support both cases(catch notfound error internally and use different endpoint)

Reply:
whoami should work with external OIDC